File: src/main/scala/com/celestia/models/GLProgram.scala

package com.celestia.models


/**
  * Class for holding data to do with GL Programs. 
  * This class holds ID's pointing to a valid GL
  * program, and the shaders that construct it. 
  */
case class GLProgram(
  programId:Int,
  vshaderId:Int,
  fshaderId:Int,
  gshaderId:Int
)

File: src/main/java/com/celestia/models/GLProgram.java

package com.celestia.models;

import java.util.List;

public class GLProgram {
    public final int programId;
    public final List<Integer> shaders;
    
    public GLProgram(int programId, List<Integer> shaders) {
        this.programId = programId;
        this.shaders = shaders;
    }
}
